ga4gh/datamodel/reads.py

```diff
@@ -232,15 +233,15 @@ def getReadAlignments(self, referenceId=None, start=None, end=None):
         """
         # TODO If referenceId is None, return against all references,
         # including unmapped reads.
-        referenceName = ""
         samFile = self.getFileHandle(self._samFilePath)
-        if referenceId is not None:
-            self.sanitizeGetRName(referenceId)
-            referenceName = samFile.getrname(referenceId)
-        referenceName, start, end = self.sanitizeAlignmentFileFetch(
-            referenceName, start, end)
+        referenceId, start, end = self.sanitizeAlignmentFileFetch(
+            referenceId, start, end)
         # TODO deal with errors from htslib
-        readAlignments = samFile.fetch(referenceName, start, end)
+        try:
+            readAlignments = samFile.fetch(referenceId, start, end)
+        except ValueError:
```

Comment:
yeah, good point.  the remedy will involve moving some pysam-like logic up into our code, but I guess the alternative (catching a generic ValueError) is worse.